sentry-ruby/lib/sentry/configuration.rb

```diff
@@ -386,7 +386,19 @@ def server_name_from_env
       end
     end
 
-    # allow extensions to extend the Configuration class
-    def post_initialization_callback; end
+    def post_initialization_callback
+      @@post_initialization_hooks ||= []
+      @@post_initialization_hooks.each do |hook|
+        instance_eval(&hook)
+      end
+    end
+
+    protected
+
+    # allow extensions to add their hooks to the Configuration class
+    def self.add_post_initialization_hook(&block)
```

Comment:
there should also be a class method called `post_initialization_callbacks` for retrieving the callbacks.